frontend/src/GroceryList.js

```diff
@@ -635,16 +681,22 @@ class GroceryListItem extends Component{
                     />
                 </Grid>
                 <Grid item xs sm className={this.props.classes.inputGrid}>
-                    <TextField
-                        variant="outlined"
-                        className={textFieldClass}
-                        id={`${this.props.ingredient}_unit`}
-                        onChange={this.handleChange}
-                        onKeyPress={this.handleKeyPress}
-                        value={this.props.unit}
-                        size='small'
-                        // style={{float: "left", clear: 'both'}}
-                    />
+                    <Select value={this.props.unit}
+                            id={`${this.props.ingredient}_unit`}
+                            onChange={this.handleUnitChange}
+                            variant="outlined"
+                            className={textFieldClass + ' ' + this.props.classes.select}
+                            >
+                        {ALL_UNITS.map((unit) => <MenuItem value={unit} 
+                                                            key={unit} 
+                                                            id={`${this.props.ingredient}_unit`}
+                                                            style={{
+                                                                // Allow units in ALL_UNITS, but only those in VISIBLE_UNITS are visible in selection menu
```

Comment:
This is not working. When I input 1 kg of rice, for example, it's giving me a blank.

What I would recommend is instead of a "Select", we use a "Autocomplete" - see addReco/addRecoApp.js for an example - and keep all the units.

Reply:
Good catch. It's not working because I didn't add all units to ALL_UNITS; it'll work if you add 'kg' to the array. Basically, if we copy all unit variations from https://github.com/ghgindex/ghgi/blob/main/ghgi/parser.py#L9 that GHGI might return then it should work with all units that work with GHGI. Here, GHGI returns 'kg' but it shows blank because 'kg' is not an item in ALL_UNITS. So, it's an easy fix.

But I suppose Autocomplete is nice since you can type in input....

If the Select ends up working, would you say Autocomplete is still better?

Reply:
I would say so since it's the best of both worlds, and it shouldn't be too difficult to switch. :) I would recommend then that we only use ALL_UNITS. As it is implemented now, even if we input "kilogram" rather than "kg" in the initial search, it still won't show up because you are limiting display to only those in VISIBLE_UNITS.

If you really want to shorten the list (I don't think it's worth the effort), refer to my addReco/addRecoApp.js, basically even if you input "kilogram" you can still autocomplete to "kg". This is done by a combination of the getOptionLabel prop and the getOptionLabel prop.

Reply:
OK, I'll convert it to Autocomplete then, when I have time.

Should the autocomplete list include abbreviated versions of each unit (eg. kg), full versions of each unit (eg. kilogram), or both? 

BTW, did you pull latest changes to test "kilogram"? I only set it to not display in the menu selection, but it should display if GHGI returns a unit that matches the unit in ALL_UNITs. In this case, the parser returns "kg" as the unit even if you type "kilogram", so adding "kg" to ALL_UNITs should fix the issue (the issue shouldn't be limiting display since display is limited only for selectable items, not current selected item). Although this doesn't matter now.